src/styles/main.scss

```diff
@@ -0,0 +1,58 @@
+@import '../styles/variables';
+
+body {
+  margin: 0;
+  width: 100vw;
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-family: $font-family;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: calc(7 * ($day-size + 1px));
+  padding: $padding;
+  &__day {
+    position: relative;
+    width: $day-size;
+    height: $day-size;
+    border: 1px solid black;
+    box-sizing: border-box;
+    background-color: $day-color;
+    transition: all 0.5s ease;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+        position: absolute;
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
+        font-size: $day-font-size;
+      }
+    }
+  }
+  &__day:hover {
+    cursor: pointer;
+    background-color: #ffbfcb;
+    transform: translate(0, -20px);
```

Comment:
Using `translate` for hover effects can lead to performance issues on low-powered devices. Consider using `transform: scale()` for better performance, especially if you're animating many elements.